runtime/drivers/duckdb/extensions.go

```diff
@@ -0,0 +1,134 @@
+package duckdb
+
+import (
+	"compress/gzip"
+	"database/sql"
+	"embed"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Since DuckDB is called from multiple packages, the extensions are installed in the init function
+func init() {
+	err := installExtensions()
+	if err != nil {
+		// If extensions cannot be installed, log the error and continue as the extensions can be downloaded
+		// Should it be fatal in order to notice the issue prior to a release?
+		log.Printf("Error preparing DuckDB extensions: %v", err)
+	}
+}
```

Comment:
I think this might delay process start. Can we move this to lazy evaluation, e.g. using a `sync.Once` global that is called when the first DuckDB handle is opened?

Reply:
Do you expect something similar to this?
```
var once sync.Once
func (c *connection) prepareExtensions() error {
	var err error
	once.Do(func() {
		err = installExtensions()
	})
	return err
}
```
However DuckDB is also used in `duckdbsql` package (`query` [function](https://github.com/rilldata/rill/blob/main/runtime/pkg/duckdbsql/duckdbsql.go#L42)). This function is called on app start by `emitStartEvent`.

Reply:
Yeah something like that. You could make it a public function, like `duckdb.PrepareExtensions`, and call it from the `duckdbsql` package before loading an extension.

I understand in practice it gets loaded fast for `rill start`, but there are other CLI commands like `rill user add` etc. that don't need DuckDB, and also some tests that don't need it but might still load the DuckDB driver.

Reply:
Moved to separate package to resolve a circular dependency and also I decided to ignore the case if no extensions are embedded at all as this might be noisy in dev runs and tests